Mathlib/Order/Filter/SmallSets.lean

```diff
@@ -54,6 +54,14 @@ theorem hasBasis_smallSets (l : Filter α) :
     HasBasis l.smallSets (fun t : Set α => t ∈ l) powerset :=
   l.basis_sets.smallSets
 
+theorem Eventually.exists_mem_basis_of_smallSets {p : ι → Prop} {s : ι → Set α} {P : Set α → Prop}
+    (h₁ : ∀ᶠ t in l.smallSets, P t) (h₂ : HasBasis l p s) : ∃ i, p i ∧ P (s i) :=
+  (h₂.smallSets.eventually_iff.mp h₁).imp fun _i ⟨hpi, hi⟩ ↦ ⟨hpi, hi Subset.rfl⟩
```

Comment:
Is the converse statement,
```lean
theorem Frequently.smallSets_of_forall_mem_basis {p : ι → Prop} {s : ι → Set α} {P : Set α → Prop}
    (h₁ : ∀ i, p i → P (s i)) (h₂ : HasBasis l p s) : ∃ᶠ t in l.smallSets, P t :=
  h₂.smallSets.frequently_iff.mpr fun _ hi => ⟨_, Subset.rfl, h₁ _ hi⟩
```
also worth having?

Reply:
I don't know how to formulate the first one to make it useful. For the second one, the useful form is `∃ᶠ t in l.smallSets, t ∈ l`, and we already have it: `Filter.frequently_smallSets_mem`.

Reply:
More details: I mostly use `(h : Frequently _ _)` to do something like `(h.and_eventually _).exists`, and `Eventually.exists_mem_basis_of_smallSets ` is enough for this use case. If you have other use cases in mind, then we can add `Frequently.smallSets_of_forall_mem_basis`.

Reply:
My thinking was less that these lemmas are obviously useful, and more that they are obviously analogous to the lemmas which _are_ useful, and having them in mathlib saves the reader from wondering what they would look like.

Reply:
For the first lemma, it's OK to add it, though I don't expect it to be used anytime soon. For the second lemma, I would prefer to have a comment explaining that `Filter.frequently_smallSets_mem` is the useful `Frequently` counterpart.

Reply:
I've added one and a comment in place of the other.

Should we `maintainer merge` this?